src/errors.rs

```diff
@@ -0,0 +1,36 @@
+
+use std::error::Error;
+use std::fmt;
+
+// TODO: distinguish between UnknownLicenseId and UnknownExceptionId
+// after https://github.com/nikomatsakis/lalrpop/issues/113 is resolved
+
+#[derive(Debug, Clone, Copy)]
+pub enum SpdxParseError<'a> {
+    UnknownLicenseId(&'a str),
+    UnknownExceptionId(&'a str),
+    InvalidStructure(&'a str),
+}
+
+impl<'a> fmt::Display for SpdxParseError<'a> {
+    fn fmt(&self, format: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        match *self {
+          SpdxParseError::UnknownLicenseId(info)
+              => format.write_fmt(format_args!("{}: {}", self.description(), info)),
```

Comment:
The conventional way to write this would be `write!(format, "{}: {}", self.description(), info)`, which expands to the same thing, but is easier to read. I don't know of a case where its considered preferable to use `format_args!` directly; there are macros which wrap it for every occasion.